### PR TITLE
Feat/requirements proxy

### PR DESCRIPTION
**Pull request by IanAtCredo**
## Change description
Two main things in this PR:

* A "max proxy score" was introduced to create a metric that can be used for fair.1: proxy detection
* Functions were set up to infer model type/framework and target type. This info can then be brought into assessment requirements ensuring (for example) that assessments that only run on binary classification models are appropriately identified.

Model_type is somewhat overlapping with other requirements (e.g., function descriptors). For instance, we know that a model that has a `generator_fun` is a language model. But other times it is ambiguous, and those "other times" are quite often! `predict` is not enough!

### CODE REVIEW

File: credoai/utils/model_utils.py [hunk 1]
```diff
@@ -17,10 +19,28 @@ def get_generic_classifier():
             model = RandomForestClassifier()
         return model
 
+
+def get_model_info(model):
```
Comment (amrasekh) [inline credoai/utils/model_utils.py:23]: For Keras and Torch, we can probably infer whether it is classification or regression through looking at the labels in the dataset.

**Review by amrasekh — APPROVED** [review]
Made some suggestions.

File: credoai/utils/constants.py [hunk 2]
```diff
@@ -9,4 +9,8 @@
     'fair.4': 'Fairness',
     'fair.5': 'Fairness',
     'fair.7': 'Fairness'
-}
\ No newline at end of file
+}
+
+MODEL_TYPES = ['TABULAR_CLASSIFIER', 'TABULAR_REGRESSOR', 'NEURAL_NETWORK']
```
[the remark this answers is not in the document]
Comment (IanAtCredo) [reply credoai/utils/constants.py:14]: Good point. Like your point about `data_type` too

[the remark this answers is not in the document]
Comment (IanAtCredo) [reply credoai/utils/constants.py:14]: Unfortunately, it's not obvious to me right now how to infer whether a neural network is a classifier or regressor. I think it's okay for them to not be mutually exclusive (though not ideal). This list will evolve over time as we figure out what distinctions we need.

File: credoai/utils/model_utils.py (hunk 1 above)
Comment (IanAtCredo) [reply credoai/utils/model_utils.py:23]: Unfortunately, that's a function of the data and the model. We actually may need the objective function - a 10 label dataset could be viewed as an ordinal regression or a classification.
